lib/rubocop/cop/sorbet/valid_sorbet_sigil.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop checks that every Ruby file contains a valid Sorbet sigil.
+      # Adapted from: https://gist.github.com/clarkdave/85aca4e16f33fd52aceb6a0a29936e52
+      #
+      # @example
+      #
+      #  # bad
+      #  # (start of file)
+      #  class Foo; end
+      #
+      #  # bad
+      #  # (start of file)
+      #  # typed: no
+      #
+      #  # good
+      #  # (start of file)
+      #  # typed: true
+      class ValidSorbetSigil < RuboCop::Cop::Cop
+        def investigate(processed_source)
+          return if processed_source.tokens.empty?
+
+          sorbet_sigil_line = sorbet_typed_sigil_comment(processed_source)
+
+          if sorbet_sigil_line.nil?
+            token = processed_source.tokens.first
+
+            add_offense(
+              token,
+              location: token.pos,
+              message: 'No Sorbet sigil found in file. ' \
+                'Try a `typed: false` to start (you can also use `rubocop -a` to automatically add this).'
+            )
+          else
+            strictness = sorbet_typed_strictness(sorbet_sigil_line)
+            return if valid_sorbet_strictness?(strictness)
+
+            add_offense(
+              sorbet_sigil_line,
+              location: sorbet_sigil_line.pos,
+              message: "Invalid Sorbet sigil `#{strictness}`."
+            )
+          end
+        end
+
+        def autocorrect(_node)
+          lambda do |corrector|
+            return unless sorbet_typed_sigil_comment(processed_source).nil?
+
+            token = processed_source.tokens.first
+
+            corrector.insert_before(token.pos, "# typed: false\n")
+          end
+        end
+
+        private
+
+        SORBET_SIGIL_REGEX = /#\s+typed:\s+([\w]+)/
+
+        def sorbet_typed_sigil_comment(processed_source)
+          processed_source.tokens
+            .take_while { |token| token.type == :tCOMMENT }
```

Comment:
What if there are empty lines before the comments? I think that's still valid, no? But I assume this code will not catch it.

Reply:
Sorbet allows the `typed` sigil to be anywhere in the document.

But maybe this cop should enforce that the sigil is in the first lines (before or after the `frozen_string_literal`)

Reply:
It ignores whitespace. I've added a test to demonstrate this.

Reply:
Ohhh, I see the test now. Sorry, totally missed it in my first reading, even though I looked for it. My bad, this looks good.

Reply:
Funny anecdote here: We have a test file in tapioca that starts like this:
```ruby
# frozen_string_literal: true

require "spec_helper"
require "pathname"
require "shellwords"

module Contents
  FOO_RBI = <<~CONTENTS
    # This file is autogenerated. Do not edit it by hand. Regenerate it with:
    #   generate command

    # typed: true

    module Foo
      def self.bar(a = _, b: _, **opts); end
    end

    Foo::PI = T.let(T.unsafe(nil), Float)
  CONTENTS
...
```

When I ran `srb init` the other day, it actually changed the `# typed: true` in there to `# typed: false` 😄

Reply:
Something similar might be a good test to add, btw:
```ruby

# frozen_string_literal: true

SOMETHING = <<~RUBY
  # typed: true
  Some Ruby code
RUBY
```
kind of thing. For this file, the cop should complain that it is NOT typed.

Reply:
Added a test for that!